tests/installation/add_maintenance_repos.pm

```diff
@@ -0,0 +1,37 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Add new add-ons using maintenance test repo URLs.
+#
+# Maintainer: QA SLE YaST team <qa-sle-yast@suse.de>
+
+use base 'y2_installbase';
+use strict;
+use warnings;
+use testapi;
+
+sub run {
+    my @repos = split(/,/, get_var('MAINT_TEST_REPO'));
+
+    $testapi::distri->get_add_on_product()->confirm_like_additional_add_on();
+
+    my $maintrepo = shift @repos;
+    $testapi::distri->get_add_on_product()->accept_current_media_type_selection();
+    $testapi::distri->get_repository_url()->add_repo({url => $maintrepo});
+    save_screenshot;
+
+    foreach (@repos) {
+        $testapi::distri->get_add_on_product_installation()->add_add_on_product();
+        $testapi::distri->get_add_on_product()->accept_current_media_type_selection();
+        my $repo = $_;
```

Comment:
this is not usually done like that, you can just `for my $repo (@repos)`.
always avoid to declare variable, because Perl is thought to be more expressive (except when it is too cryptic, of course).

Reply:
done